### PR TITLE
added the option to build boost with multi-thread support (thread saf…

### PR DESCRIPTION
**Pull request by mboisson**
Added the option to enable boost multi-thread support.

### CODE REVIEW

File: easybuild/easyblocks/b/boost.py [hunk 1]
```diff
@@ -193,6 +194,36 @@ def build_step(self):
...
+            bjammpimtoptions = "%s  --user-config=user-config.jam --with-mpi threading=multi --layout=tagged" % bjamoptions
+
+            # build mpi multi-thread lib
+            run_cmd("./bjam %s %s" % (bjammpimtoptions, paracmd), log_all=True, simple=True)
+
+            # install the multi-thread version of boost
+            run_cmd("./bjam %s  install %s" % (bjammpimtoptions, paracmd), log_all=True, simple=True)
+            
+            # cleanup previous build before proceeding with the full Boost
+            run_cmd("./bjam --clean-all", log_all=True, simple=True)
```
Comment (boegel) [inline easybuild/easyblocks/b/boost.py]: @mboisson This is a bit too copy-pasty...

We should define a separate method and reuse that (note: untested code below):

```python

def build_boost_variant(self, bjamoptions):
    """Build Boost library with specified options for bjam."""
    # build with specified options
    run_cmd("./bjam %s %s" % (bjamoptions, paracmd), log_all=True, simple=True)
    # install built Boost library
    run_cmd("./bjam %s install %s" % (bjamoptions, paracmd), log_all=True, simple=True)
    # clean up before proceeding with next build
    run_cmd("./bjam --clean-all", log_all=True, simple=True)

def build_step(self):
    ...
    if self.cfg['boost_mpi']:
        self.log.info("Building Boost MPI library")
        self.build_boost_variant(bjamoptions + "--user-config=user-config.jam --with-mpi")

    if self.cfg['boost_multi_thread']:
        self.log.info("Building Boost multi-threading library")
        self.build_boost_variant(bjamoptions + "threading=multi --layout=tagged")

    if self.cfg['boost_multi_thread'] and self.cfg['boost_mpi']:
        self.log.info("Building Boost MPI library with multi-threading")
        self.build_boost_variant(bjamoptions + "--user-config=user-config.jam --with-mpi threading=multi --layout=tagged")
```

Comment (mboisson) [inline easybuild/easyblocks/b/boost.py]: Moved repetitive code into a function.

File: easybuild/easyblocks/b/boost.py [hunk 2]
```diff
@@ -181,18 +191,16 @@ def build_step(self):
...
-
-            # boost.mpi was built, let's 'install' it now
-            run_cmd("./bjam %s  install %s" % (bjammpioptions, paracmd), log_all=True, simple=True)
-            
-            # cleanup previous build before proceeding with the full Boost
-            run_cmd("./bjam --clean-all", log_all=True, simple=True)
+        # if both boost_mpi and boost_multi_thread are enabled, build boost mpi with multi-thread support
+        if self.cfg['boost_multi_thread'] and self.cfg['boost_mpi']:
+            self.log.info("Building boost_mpi with multi threading")
+            self.build_boost_variant(bjamoptions + " --user-config=user-config.jam --with-mpi threading=multi --layout=tagged",paracmd)
```
Comment (boegel) [inline easybuild/easyblocks/b/boost.py]: @mboisson lines should be kept under 120 chars, so:

```python
if self.cfg['boost_multi_thread'] and self.cfg['boost_mpi']:
    self.log.info("Building boost_mpi with multi threading")
    extra_bjamoptions = " --user-config=user-config.jam --with-mpi threading=multi --layout=tagged"
    self.build_boost_variant(bjam_options + extra_bjamoptions, paracmd)
```


also, style nitpicking: space after `,` (before `paracmd`)

Comment (mboisson) [inline easybuild/easyblocks/b/boost.py]: corrected

**Comment by boegel** [comment]
Tested on my end, works as expected, thanks @mboisson!

**Pull request merged**